tests/unit/main.c

```diff
@@ -399,20 +400,40 @@ ctest_is_unit_test(const char *argv0)
    return 0;
 }
 
-// Determine the log message verbosity to use for this test run.
-// If env var VERBOSE is unset, or is set to "0" or "false", the be quiet.
-// Otherwise, be verbose.
-static bool
+/*
+ * Determine the log message verbosity level to use for this test run.
+ *
+ * If env var VERBOSE is unset, or is set to "0" then be quiet.
+ * If env var VERBOSE=1 be verbose reporting only errors.
+ * Otherwise, settle for the specified verbosity level.
+ *
+ * NOTE: Currently, the only other level we support is level=7 to print all
+ *       debugging output. Unknown verbosity levels default to silent output.
+ *
+ * As an example, for debugging, to see outputs from a test case that generates
+ * diagnostic output, invoke as:
+ *
+ *   $ VERBOSE=7 bin/unit/splinter_test test_splinter_print_diags
+ */
+static msg_level
 ctest_get_verbosity()
 {
    char *val = getenv("VERBOSE");
    if (val == NULL) {
-      return 0;
+      return MSG_LEVEL_SILENT;
    }
-   if ((strcmp(val, "0") == 0) || (strcmp(val, "false") == 0)) {
-      return 0;
+   if (STRING_EQUALS_LITERAL(val, "0")) {
+      return MSG_LEVEL_SILENT;
    }
-   return 1;
+
+   if (STRING_EQUALS_LITERAL(val, "1")) {
+      return MSG_LEVEL_ERRORS;
+   }
+   if (STRING_EQUALS_LITERAL(val, "7")) {
+      return MSG_LEVEL_DEBUG;
+   }
+   // Unknown verbosity level; fall-back to silent output mode.
+   return MSG_LEVEL_SILENT;
```

Comment:
Is it much harder to use `atoi` or the modern secure equivalent?  This "exact-match only with fallback" behavior is pretty confusing e.g. `VERBOSE=3`  is the same as `VERBOSE=0`.

Reply:
I actually had coded it up to use `atoi` in an earlier version.

But then I felt `VERBOSE=3` does nothing as that is an unsupported verbosity level. So, supporting that (or even allowing it thru this interface) appeared confusing by itself to me.

Which is why I went the route of only 'honoring' verbosity levels we know of now and support. The remaining unknown ones default (fall-back) to silent.

This way, one day when we do fill out the `MSG_LEVEL_<>` stream, the usage of `VERBOSE=4` will properly mean and map to something supported.